boards: nxp_rw612: add  u-blox EVK-IRIS-W1x support

3rang opened the pull request:
**boards: add u-blox EVK-IRIS-W1 (RW612) support**

This PR adds support for the u-blox EVK-IRIS-W1 evaluation board based on the NXP RW612 chipset.

**Added files:**
- Board configuration: `CMakeLists.txt`, `board.cmake`, `pre_dt_board.cmake`, `board.yml`
- Kconfig: `Kconfig.defconfig`, `Kconfig.ubx_evk_iris_w1`
- Device tree: `ubx_evk_iris_w1_rw612.dts`, `ubx_evk_iris_w1_common.dtsi`, `ubx_evk_iris_w1_rw612-pinctrl.dtsi`, `ubx_evk_iris_w1_rw612.yaml`
- Defconfig: `ubx_evk_iris_w1_rw612_defconfig`
- Flash config: `fidelex_flash_config.c`, `macronix_flash_config.c`
- Documentation: `doc/index.rst`, `doc/EVK-IRIS-W1.jpg`

**Tested:**  
- Built and ran Zephyr samples (`hello_world`, `blinky`) on EVK-IRIS-W1 hardware.
- Verified UART and basic peripheral functionality.

``` 
*** Booting Zephyr OS build v4.1.0-3142-g039861cb36d0 ***
Hello World! ubx_evk_iris_w1/rw612
```
**References:**
https://www.u-blox.com/en/product/evk-iris-w1

matt-wood-ct commented:
How do the RF limits get applied for this module? is that already included in the NXP HAL?

3rang commented:
> How do the RF limits get applied for this module? is that already included in the NXP HAL?

Hi @matt-wood-ct , 

This PR adds initial support for the EVK-IRIS-W1x board to get it listed in Zephyr. Region-specific RF configuration files are currently being tested and will be published in the [u-blox GitHub repository](https://github.com/u-blox/u-blox-sho-OpenCPU). 
For now, we can use them by copying the files into your project manually. 

Once finalized, we plan to add them to Zephyr. Not sure — would it make sense to handle this via the NXP HAL?

matt-wood-ct commented:
> Once finalized, we plan to add them to Zephyr. Not sure — would it make sense to handle this via the NXP HAL?

It almost certainly shouldn't live in there, I only mention that because NXP already have several modules RF limits present in this path  
`hal/nxp/mcux/mcux-sdk-ng/components/wifi_bt_module`
The files in there offer some global structs: `tx_pwrlimit_2g_cfg` and `tx_pwrlimit_5g_cfg` which are used to configure the power limits.
The headers which do this are pulled in by this cmake file 
https://github.com/zephyrproject-rtos/hal_nxp/blob/04b5aed4d6ebf2ae4e3eea34a703fe16824a3459/mcux/middleware/wifi_nxp/CMakeLists.txt

They are using an odd trick I haven't seen before, they use a `#define` macro value inside a `#include` here  
https://github.com/zephyrproject-rtos/hal_nxp/blob/db164967df1fd234401e854acedccae387877f91/mcux/middleware/wifi_nxp/wlcmgr/wlan_txpwrlimit_cfg.c#L18

dleach02 commented:
We have a reference PR that will address the RF limits. See the following for how we are proposing to deal with this:

 #88951

3rang commented:
Hi @kartben and  (NXP Team) @dleach02 , @mmahadevan108

I've resolved the CI issue and Quality Gate passed by Sonar bot with 0 issues. 
Just checking if there's anything else required from my side, or if this is ready for final review/merge.
Thanks!

-Tarang

dleach02 commented:
> > How do the RF limits get applied for this module? is that already included in the NXP HAL?
> 
> Hi @matt-wood-ct ,
> 
> This PR adds initial support for the EVK-IRIS-W1x board to get it listed in Zephyr. Region-specific RF configuration files are currently being tested and will be published in the [u-blox GitHub repository](https://github.com/u-blox/u-blox-sho-OpenCPU). For now, we can use them by copying the files into your project manually.
> 
> Once finalized, we plan to add them to Zephyr. Not sure — would it make sense to handle this via the NXP HAL?

@3rang our PR #88951 merged so you now have a method to have your own files in your board folder.

3rang committed:
boards: nxp_rw612: Add support for u-blox EVK-IRIS-W1x

Add basic board definition of the u-blox EVK-IRIS-W1-rw612 board.
Adds maintainer info for u-blox boards.

Signed-off-by: Tarang Patel <tarang3399.patel@gmail.com>

kartben commented on boards/u-blox/ubx_evk_iris_w1/doc/index.rst:
```diff
@@ -0,0 +1,124 @@
+.. note::
+   Board: ubx_evk_iris_w1
```
Please follow board template -- https://github.com/zephyrproject-rtos/zephyr/blob/75d50742a710cf62112cb81fe0323fcc7b9cabde/doc/templates/board.tmpl

3rang replied:
i will fix it in next push.

kartben replied:
A reminder to please check you're using the board template. For example, using board:supported-hw and board:supported-runners directives

kartben replied:
Yes, please :)

kartben commented on boards/u-blox/ubx_evk_iris_w1/doc/index.rst:
```diff
@@ -0,0 +1,124 @@
+
+Overview
+********
+
+.. figure:: EVK-IRIS-W1.jpg
+   :align: center
+   :alt: EVK-IRIS-W1
```
```suggestion
```

kartben replied:
this hasn't been addressed

kartben commented on boards/u-blox/ubx_evk_iris_w1/doc/index.rst:
```diff
@@ -0,0 +1,124 @@
+Macronix. To ensure proper operation, you must configure the software to match the flash memory
+type present on your board.
+
+This directory contains configuration files for each supported flash memory. You can select the
+appropriate configuration by setting one of the following options in your
+``ubx_evk_iris_w1_rw612_defconfig`` file:
+
+- ``CONFIG_FLASH_FIDELEX``
+- ``CONFIG_FLASH_MACRONIX``
+
+For module builds up to 2023 week 45, use Macronix flash. For builds from week 46 of 2023 (2346) or later, use Fidelex flash.
```
This could probably be board revisions instead? @nordicjm

3rang replied:
we have EVK with two different flash config.

3rang commented:
@kartben , could you please take a look at this PR? It has passed all checks.

kartben commented on boards/u-blox/ubx_evk_iris_w1/doc/index.rst:
```diff
@@ -0,0 +1,123 @@
+.. zephyr:board::ubx_evk_iris_w1
```
```suggestion
.. zephyr:board:: ubx_evk_iris_w1
```

3rang replied:
@kartben 

I've encountered an issue when adding a space. Specifically, the first documentation build fails, but re-running make html without cleaning allows the build to succeed. I suspect that the `zephyr:board` directive is initially unable to locate `ubx_evk_iris_w1`, but it resolves correctly on the subsequent build (without cleaning _build).  I tested with another board name and worked in first build. This issue seems to be caused by the space in the board name, leading to documentation build failures locally , which would also affect CI processes.

Could you assist in addressing this issue or consider allowing this change? If necessary, I can incorporate the adjustment in a subsequent PR to align with the requirements.

kartben replied:
Please push an update with the space. It should be there, and it should work :) if it doesn't, at least I can try to help by looking at the CI failure logs

kartben replied:
https://builds.zephyrproject.io/zephyr/pr/89274/docs/boards/u-blox/ubx_evk_iris_w1/doc/index.html
:)

kartben reviewed (dismissed):
+1 for docs, thanks!

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/CMakeLists.txt:
```diff
@@ -0,0 +1,29 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+zephyr_library()
```
this needs to be used only when a file is added, below needs to be reworked

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/CMakeLists.txt:
```diff
@@ -0,0 +1,29 @@
+     "Define either CONFIG_FLASH_MACRONIX or CONFIG_FLASH_FIDELIX.")
+  endif()
+
+if (CONFIG_DT_HAS_NXP_ENET_MAC_ENABLED AND CONFIG_XTAL32K)
```
missing indent, also wrongly has space between if and bracket in cmake

3rang replied:
ok, Noted.

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/Kconfig.defconfig:
```diff
@@ -0,0 +1,14 @@
+if BOARD_UBX_EVK_IRIS_W1
+
+if COUNTER_MCUX_LPC_RTC_1HZ
+config XTAL32K
+	default y
```
missing newlines before and after

3rang replied:
ok, Noted.

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/Kconfig.ubx_evk_iris_w1:
```diff
@@ -0,0 +1,11 @@
+config BOARD_UBX_EVK_IRIS_W1
+	select SOC_PART_NUMBER_RW612ETA2I
+
+config FLASH_MACRONIX
+	bool "Macronix Flash support"
+config FLASH_FIDELEX
+	bool "Fidelex Flash support"
```
this should not even be a Kconfig, use board variants if both are on the board or board revision if there's 2 different board hardware revisions with the different devices

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/ubx_evk_iris_w1_common.dtsi:
```diff
@@ -0,0 +1,229 @@
+	// IRIS-W1 RGB
+	rgb_leds {
+		compatible = "gpio-leds";
+		rgb_blue: led_1 {
```
newline here (line 39) and between each node, fix in whole PR

3rang replied:
ok, Noted.

nordicjm replied:
not fixed

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/ubx_evk_iris_w1_common.dtsi:
```diff
@@ -0,0 +1,229 @@
+		zephyr,uart-mcumgr = &flexcomm3;
+	};
+
+
```
```suggestion
```

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/ubx_evk_iris_w1_rw612_defconfig:
```diff
@@ -0,0 +1,17 @@
+# Copyright (c) 2025 u-blox AG
+# SPDX-License-Identifier: Apache-2.0
+
+
+
+
```
```suggestion
```

3rang replied:
ok, Noted.

3rang commented on boards/u-blox/ubx_evk_iris_w1/ubx_evk_iris_w1_common.dtsi:
```diff
@@ -0,0 +1,229 @@
+			zephyr,code = < 0x3 >;
+		};
+	};
+
```
ok, Noted.

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/CMakeLists.txt:
```diff
@@ -0,0 +1,22 @@
+
+if(CONFIG_NXP_RW6XX_BOOT_HEADER)
+	zephyr_compile_definitions(BOARD_FLASH_SIZE=${CONFIG_FLASH_SIZE}*1024)
+	if (BOARD_REVISION STREQUAL "macronix")
```
no space between if and bracket in cmake

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/Kconfig.defconfig:
```diff
@@ -0,0 +1,16 @@
+if COUNTER_MCUX_LPC_RTC_1HZ
+
+config XTAL32K
+default y
```
missing indent

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/CMakeLists.txt:
```diff
@@ -0,0 +1,22 @@
+	endif()
+
+	if(CONFIG_DT_HAS_NXP_ENET_MAC_ENABLED AND CONFIG_XTAL32K)
+	message(FATAL_ERROR "Ethernet and external 32K clock source are "
```
missing indent

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/ubx_evk_iris_w1_common.dtsi:
```diff
@@ -0,0 +1,229 @@
+
+	sw {
+		compatible = "gpio-keys";
+		user_button_0: sw_1 {
```
nor here

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/ubx_evk_iris_w1_common.dtsi:
```diff
@@ -0,0 +1,229 @@
+	rx-clock-source = <1>;
+	rx-clock-source-b = <1>;
+	/* external flash */
+	w25q512jvfiq: w25q512jvfiq@0 {
```
nor here

nordicjm commented on boards/u-blox/ubx_evk_iris_w1/ubx_evk_iris_w1_common.dtsi:
```diff
@@ -0,0 +1,229 @@
+				label = "mcuboot";
+				reg = <0x00000000 DT_SIZE_K(128)>;
+			};
+			slot0_partition: partition@20000 {
```
nor here, etc. fix in whole PR

dleach02 reviewed (changes requested):
@kartben, should we use this PR to add an entry in the maintainers.yml to assign responsibility for support to someone for the growing list of u-blox boards (presumably 3rang)? Otherwise this will become a floating board without clear responsibility.

See https://github.com/zephyrproject-rtos/zephyr/blob/24f9ca7178a13273844bbe710a0bb37db179ad03/MAINTAINERS.yml#L2409 as a reference

kartben commented:
> @kartben, should we use this PR to add an entry in the maintainers.yml to assign responsibility for support to someone for the growing list of u-blox boards (presumably 3rang)? Otherwise this will become a floating board without clear responsibility.
> 
> See
> 
> https://github.com/zephyrproject-rtos/zephyr/blob/24f9ca7178a13273844bbe710a0bb37db179ad03/MAINTAINERS.yml#L2409
> 
> as a reference

@dleach02 

if @3rang is willing to step up as a maintainer, yes, by all means, but otherwise there is no strict requirement that boards should be covered in MAINTAINERS file (for better or worse), so please consider not blocking the PR if this is the only thing for which you are requesting changes. Thanks!

kartben commented:
no trailing wildcard in the glob pattern
also please don't add to the end of the file but rather circa line 4670 to preserve alphabetical order (i.e after Tracing and before USB)

```
u-blox Platforms:
  status: maintained
  maintainers:
    - 3rang
  files:
    - boards/u-blox/
  labels:
    - "platform: u-blox"
```

dleach02 commented:
@nordicjm can you reapply your +1. We asked @3rang to be a maintainer over the u-blox boards so he updated things with this change.
